sdk/log/record.go

```diff
@@ -233,3 +233,20 @@ func (r *Record) Clone() Record {
 	res.back = slices.Clone(r.back)
 	return res
 }
+
+// TestRecord is a log record for testing.
+// You can use it for unit testing [Processor] and [Exporter] implementations.
+// Do not use TestRecord in production code.
+type TestRecord struct {
```

Comment:
Maybe it would be better to have such think it `sdk/log/logtest` package. I have not tried implementing it, because I think it will require a lot of boilerplate and I am not sure if it is worth (trade-offs).

I think that @MrAlias also was thinking about it (he believe he mentioned it during the SIG)